### ISSUE TITLE
Keeping device profiles active so Google doesn't think they're not used and mark them inactive? 

### ISSUE THREAD

**Issue by archon810** [NoMore201/googleplay-api#39]
I've been using [DummyDroid](http://www.onyxbits.de/dummydroid) to create fake profiles for various devices for years, and recently I noticed some of them stopped working in gplacli, and they also stopped showing up in the dropdown when logged into Google Play on the web.

For example, I was no longer able to download https://play.google.com/store/apps/details?id=com.samsung.android.email.provider with a Samsung profile.

I re-registered the same device specs with DummyDroid, got a new GSFID, and that profile worked right away downloading that Samsung APK. 

So obviously Google marks older profiles obsolete, and even downloading using them (with gplaycli) isn't enough to keep them active. 

Any idea what API call makes a profile active and keeps it active? A no-op of sorts that tells Google - "hey I'm still alive."

Thanks!

**Comment by matlink** [comment]
What's the error you get when you can't download this app?

**Comment by archon810** [comment]
You just get an app not compatible with device error.

**Comment by archon810** [comment]
@NoMore201 Do you have any ideas here? Also, can I chat with you privately about protobufs, @matlink directed me to you for possible help trying to get protoc and Charles Proxy working with Google Play API's protobufs.

Thanks!

**Comment by archon810** [comment]
I gave up on this and now we detect when the token expired and reregister for a new GSF ID.

cc @SeBsZ

**Comment by ZenBarrier** [comment]
@archon810 
How do you detect when tokens are expired?

**Comment by NoMore201** [comment]
@archon810 I'm not sure I understood your problem. Basically you have different profiles with different devices using the same google account, and when you use gpapi the new login invalidates the other profiles, right?

Regarding using a proxy with play store, you can contact me. I'm planning to write a little tutorial to setup mitmproxy to sniff play store traffic

**Comment by archon810** [comment]
@NoMore201 Not exactly. We've been using multiple profiles (GSF IDs), but at some point some of them started failing to return any APKs. When reregistered with the same set of rules (using DummyDroid), I was able to download again with these new GSF IDs.

We have now moved on to automatically refreshing and generating GSF IDs using the googleplay-api devices file (with some more devices added). 

@ZenBarrier To be honest, I don't remember anymore what the return value was for the expired GSF IDs, I just know that when I re-registered with another GSF ID, it started downloading again. But I was also using some legacy code and Play Store user agent, so maybe it was related to that.

And now that we're using the new registration method, I'm actually not sure if the profiles will expire at some point, since we only switched within the last month. If they ever stop downloading again, and it's not an expired token issue or throttling, I guess I'll see then.
